opendkim: incompatible SSL versions redux :(

I had a stock Debian 11 opendkim working kinda-sorta.
Tried building the develop version and had issues.
Uninstalled. Reinstalled stock opendkim
Now I am getting a complaint about incompatible SSL versions.
It says OpenDKIM was compiled to 0x0101010ef but 0x01010108f is installed. From opensslv.h.in, we see the difference is patch level 0x8 vs 0xe (1.1.1h vs 1.1.1n).

I have questions and a bug report:
**Questions**: 
1) Is it always correct that `opendkim -V` should correctly answer the question of which openSSL library version was compiled?
2) Can you think of how/why/where I am getting the above complaint on this system, when every binary I can find, for both openSSL and opendkim, claim to be compiled and/or _are_ version 1.1.1n?!!

**BUG**: the only difference between these is at the minor patch level. That makes no sense.

Why in the world would we require an exact match on patch level? Seems to me at _least_ the comparison ought to be (opendkim.c)
` if ((dkim_ssl_version()/256) != (OPENSSL_VERSION_NUMBER/256))` which would remove the final 8 bit patch number from the question. More likely, any version with same major and the minor being same or bigger ought to work, according to spec?

hello, were you able to get a permanent fix? 
having issues compiling opendkim version 2.11.0~beta2-6 with openssl 3.0.2
`checking for library containing ERR_peek_error... none required
configure: error: Cannot build shared opendkim
                                      against static openssl libraries.
                                      Configure with --disable-shared
                                      to get this working or obtain a
                                      shared libssl library for
                                      opendkim to use.`

any help is appreciated

--- a/opendkim/opendkim.c
+++ b/opendkim/opendkim.c
@@ -15969,7 +15969,7 @@ main(int argc, char **argv)
 #ifdef USE_GNUTLS
 	if (dkim_ssl_version() != GNUTLS_VERSION_NUMBER * 256)
 #else /* USE_GNUTLS */
-	if (dkim_ssl_version() != OPENSSL_VERSION_NUMBER)
+	if ((dkim_ssl_version() >> 8) != (OPENSSL_VERSION_NUMBER >> 8))
 #endif /* USE_GNUTLS */
 	{
 		fprintf(stderr,

